base/iterators.jl

```diff
@@ -473,7 +473,12 @@ zip_iteratoreltype() = HasEltype()
 zip_iteratoreltype(a) = a
 zip_iteratoreltype(a, tail...) = and_iteratoreltype(a, zip_iteratoreltype(tail...))
 
-last(z::Zip) = getindex.(z.is, minimum(Base.map(lastindex, z.is)))
+function last(z::Zip)
+    IteratorSize(z) == SizeUnknown() &&
+        throw(ArgumentError("Cannot get last element of zipped iterators of undefined lengths"))
```

Comment:
This is breaking. It might be only slightly breaking but even then it would need to be well justified. (`SizeUnknown` does not guarantee that length will throw)